src/tunnel.c

```diff
@@ -895,13 +878,42 @@ int ssl_connect(struct tunnel *tunnel)
 		}
 	}
 
+	if (!tunnel->config->insecure_ssl)
+		if (tunnel->config->min_tls <= 0)
+			tunnel->config->min_tls = TLS1_2_VERSION;
```

Comment:
(nitpick) you use TLS1_2_VERSION here inconditionally so checks in config.c might not make sense.
Do we still have any test with an openssl 0.9.8? The reaallly old 1.0.1 I could find already has this so it's fine as far as I would be willing to support explicitly, just wanted to point out the new ifdefs are redundant in config parsing code and below.